### ISSUE TITLE
Crop source slice length does not match destination slice length

### ISSUE THREAD

**Issue by seddonm1** [Cykooz/fast_image_resize#17]
I have been trying to do a crop of a `640x480` image to a `640x640` image (which is really about padding using a `CropBox { left: 0, top: 80, width: 640, height: 480 }`)

I believe there is an error in this line:
https://github.com/Cykooz/fast_image_resize/blob/main/src/image_view.rs#L402

I think this inclusive end range is incorrect (and differs from your original example https://github.com/Cykooz/fast_image_resize/issues/13#issuecomment-1435755734):

```rust
let row_range = (crop_box.left as usize)..=(right as usize);
```

and should be:
```rust
let row_range = (crop_box.left as usize)..(right as usize);
```

**Comment by Cykooz** [comment]
> I have been trying to do a crop of a 640x480 image to a 640x640 image

You can't crop into size bigger than size of source image. 
If you want to store resizing result into some region of destination image, you must:
- create a big destination image (640x640);
- get `DynamicImageViewMut` from it;
- crop this view into smaller (640x480) `DynamicImageViewMut`;
- use cropped `DynamicImageViewMut` as `dst_image` parameter for `Resizer::resize()` function.

**Comment by seddonm1** [comment]
Hi @Cykooz .

Sorry I was not very clear in my initial message. Here is some code to reproduce the issue:

```rust
use fast_image_resize::{CropBox, Image, PixelType, ResizeAlg, Resizer};
use std::num::NonZeroU32;

fn main() {
    let source_image = Image::new(
        NonZeroU32::new(640).unwrap(),
        NonZeroU32::new(480).unwrap(),
        PixelType::U8x3,
    );

    // Create container for data of destination image. Initialised as RGB (0,0,0)
    let mut target_image = Image::new(
        NonZeroU32::new(640).unwrap(),
        NonZeroU32::new(640).unwrap(),
        PixelType::U8x3,
    );

    // Get mutable view of destination image data
    let mut dst_view = target_image
        .view_mut()
        .crop(CropBox {
            left: 0,
            top: 80,
            width: NonZeroU32::new(640).unwrap(),
            height: NonZeroU32::new(480).unwrap(),
        })
        .unwrap();

    // do the resize
    Resizer::new(ResizeAlg::Nearest)
        .resize(&source_image.view(), &mut dst_view)
        .unwrap();
}
```

**Comment by Cykooz** [comment]
Oh, you are right. It is a bug.
I've fixed error in version 2.7.3.
Thank you.
